src/main/java/net/dv8tion/jda/api/interactions/commands/build/CommandData.java

```diff
@@ -61,10 +66,78 @@ public interface CommandData extends SerializableData
      *         True, if this command is enabled by default for everyone. (Default: true)
      *
      * @return The builder instance, for chaining
+     *
+     * @deprecated This has been deprecated in favor of {@link #setDefaultPermissions(Collection)}
      */
+    @Deprecated
+    @ForRemoval
+    @DeprecatedSince("5.0.0")
     @Nonnull
     CommandData setDefaultEnabled(boolean enabled);
 
+    /**
+     * Sets the default {@link Permission Permissions} a user must have in order to see and use this command.
+     * <br>By default, everyone can use this command.
+     * <p>Passing {@code EnumSet.noneOf(Permission.class)} will deny everyone from using this command.
+     *
+     * @param permissions Collection of {@link Permission Permissions} a user must have to execute this command.
+     *
+     * @throws IllegalArgumentException
+     *         If any of the provided Permissions are null or {@link Permission#UNKNOWN UNKNOWN}.
+     *
+     * @return The builder instance, for chaining
+     */
+    @Nonnull
+    CommandData setDefaultPermissions(@Nonnull Collection<Permission> permissions);
+
+    /**
+     * Sets the default {@link Permission Permissions} a user must have in order to see and use this command.
+     * <br>By default, everyone can use this command.
+     * <p>Passing nothing will deny everyone from using this command.
+     *
+     * @param permissions Vararg of {@link Permission Permissions} a user must have to execute this command.
+     *
+     * @throws IllegalArgumentException
+     *         If any of the provided Permissions are null or {@link Permission#UNKNOWN UNKNOWN}.
+     *
+     * @return The builder instance, for chaining
+     */
+    @Nonnull
+    default CommandData setDefaultPermissions(@Nonnull Permission... permissions)
+    {
+        return setDefaultPermissions(Arrays.asList(permissions));
+    }
+
+    /**
+     * Sets the default raw permission bitfield representing the permissions a user must have in order to see and use this command.
+     * <br>By default, everyone can use this command.
+     * <p>Passing 0 will deny everyone from using this command.
+     *
+     * @param  raw Raw permission bitfield representing the permissions a user must have to execute this command.
+     *
+     * @throws IllegalArgumentException
+     *         If raw is smaller than 0
+     *
+     * @return The builder instance, for chaining
+     */
+    @Nonnull
+    default CommandData setDefaultPermissions(long raw)
+    {
+        Checks.check(raw >= 0, "Raw permissions cannot be smaller than 0!");
```

Comment:
Permissions are unsigned, this check makes no sense.

Reply:
The user can still input any value they want